TrafficReplayer/src/org/opensearch/migrations/replay/TrafficReplayer.java

```diff
@@ -245,6 +323,44 @@ public void consumeLogLinesForFile(Path p, Consumer<ReplayEntry> replayEntryCons
         }
     }
 
+    private static String formatPacketsAsBase64String(String prefix, ArrayList<byte[]> packets) {
+        int size = packets.stream().flatMapToInt(bytearray -> IntStream.of(bytearray.length)).sum();
+        ByteBuffer packetsByteBuf = ByteBuffer.allocate(size);
+        for (byte[] packetData : packets) {
+            packetsByteBuf.put(packetData);
+        }
+        return prefix + "[" + size + "]:" + Base64.getEncoder().encodeToString(packetsByteBuf.array());
+    }
+
+    public static class TripleToFileWriter implements AutoCloseable{
+        FileOutputStream fileOutputStream;
+        BufferedOutputStream bufferedOutputStream;
+
+        public TripleToFileWriter(Path path) throws IOException {
+            fileOutputStream = new FileOutputStream(path.toFile(), true);
+            bufferedOutputStream = new BufferedOutputStream(fileOutputStream);
+        }
+
+        public void write(RequestResponseResponseJSONTriple triple) {
+            JSONObject meta = new JSONObject();
+            meta.put("request", triple.requestData);
+            meta.put("primaryResponse", triple.primaryResponseData);
+            meta.put("shadowResponse", triple.shadowResponseData);
+
+            try {
+                bufferedOutputStream.write(meta.toString().getBytes(StandardCharsets.UTF_8));
+                bufferedOutputStream.write("\n".getBytes(StandardCharsets.UTF_8));
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
Never do this, as per above - at least put lots of loud logging around it.  Eaten exceptions are a MAJOR cause of errors in java code.